test/jdk/java/awt/print/Dialog/DialogCopies.java

```diff
@@ -29,284 +29,76 @@
-        } catch (InterruptedException e) {
-            if (!testGeneratedInterrupt) throw e;
-
-            //reset flag in case hit this code more than once for some reason (just safety)
-            testGeneratedInterrupt = false;
-            if (theTestPassed == false) {
-                throw new RuntimeException(failureMessage);
-            }
+    public static void checkNoOfCopies(PrinterJob job, boolean pdReturnValue) {
+        if (pdReturnValue) {
+            System.out.println("User has selected OK/Print button on the PrintDialog");
+        } else {
+            System.out.println("User has selected Cancel button on the PrintDialog");
         }
-
-    }
-
-    public static synchronized void setTimeoutTo(int seconds) {
-        sleepTime = seconds * 1000;
-    }
-
-    public static synchronized void pass() {
-        Sysout.println("The test passed.");
-        Sysout.println("The test is over, hit  Ctl-C to stop Java VM");
-
-        // first check if this is executing in main thread
-        if (mainThread == Thread.currentThread()) {
-            /*
-             * Still in the main thread, so set the flag just for kicks,
-             * and throw a test passed exception which will be caught
-             * and end the test.
-             */
-            theTestPassed = true;
-            throw new TestPassedException();
+        int copies = job.getCopies();
+        if (copies <= 1) {
+            testResult = false;
+            System.out.println("Expected the number of copies to be more than 1 but got " + copies);
+        } else {
+            testResult = true;
+            System.out.println("Total number of copies : " + copies);
         }
-        /*
-         * pass was called from a different thread, so set the flag and interrupt the main thead.
-         */
-        theTestPassed = true;
-        testGeneratedInterrupt = true;
-        mainThread.interrupt();
+        countDownLatch.countDown();
+        instructionFrame.dispose();
     }
 
-    public static synchronized void fail() {
-        // test writer didn't specify why test failed, so give generic
-        fail("it just plain failed! :-)");
-    }
-
-    public static synchronized void fail(String whyFailed) {
-        Sysout.println("The test failed: " + whyFailed);
-        Sysout.println("The test is over, hit  Ctl-C to stop Java VM");
-        //check if this called from main thread
-        if (mainThread == Thread.currentThread()) {
-            //If main thread, fail now 'cause not sleeping
-            throw new RuntimeException(whyFailed);
+    public static void main(String[] args) throws InterruptedException {
+        countDownLatch = new CountDownLatch(1);
+        createInstructionUI();
+        showPrintDialog();
+        if (!countDownLatch.await(5, TimeUnit.MINUTES)) {
+            throw new RuntimeException("Timeout : No action was performed on the test UI.");
         }
```

Comment:
`job.printDialog()`  is a blocking call. This line will be reached only after the Print dialog is dismissed, thus the test will never time out by the latch.

The original version of the test didn't support timing out, so the updated version may skip this too. You may submit a new bug to implement timeout support.

Reply:
I will handle the timeout separately.